packages/shared/src/utils/contract/contract.tsx

```diff
@@ -122,21 +122,21 @@ export const getCancellationPrice = (contract_info: TContractInfo) => {
 
 export const isEnded = (contract_info: TContractInfo) =>
     !!(
-        (contract_info.status && contract_info.status !== 'open') ||
-        contract_info.is_expired ||
-        contract_info.is_settleable
+        (contract_info?.status && contract_info.status !== 'open') ||
+        contract_info?.is_expired ||
+        contract_info?.is_settleable
     );
 
 export const isOpen = (contract_info: TContractInfo) => getContractStatus(contract_info) === 'open';
 
-export const isUserSold = (contract_info: TContractInfo) => contract_info.status === 'sold';
+export const isUserSold = (contract_info: TContractInfo) => contract_info?.status === 'sold';
 
-export const isValidToCancel = (contract_info: TContractInfo) => !!contract_info.is_valid_to_cancel;
+export const isValidToCancel = (contract_info: TContractInfo) => !!contract_info?.is_valid_to_cancel;
 
 export const isValidToSell = (contract_info: TContractInfo) =>
-    !isEnded(contract_info) && !isUserSold(contract_info) && !!contract_info.is_valid_to_sell;
+    !isEnded(contract_info) && !isUserSold(contract_info) && !!contract_info?.is_valid_to_sell;
 
-export const hasContractEntered = (contract_info: TContractInfo) => !!contract_info.entry_spot;
+export const hasContractEntered = (contract_info: TContractInfo) => !!contract_info?.entry_spot;
```

Comment:
Do we really need question marks here? if `contract_info` is sometimes passed here as `undefined`, then the type of param should show it: `contract_info?: TContractInfo` :)

Reply:
it was throwing error like the other comment but its not anymore  🫠

Reply:
There is this error
![image](https://github.com/binary-com/deriv-app/assets/125247833/0fc44982-10ce-4997-a9f4-df72bd69463b)

Reply:
@farhan-nurzi-deriv yeah, i understand why we need '?' :) Can we also update the parameter's type in these functions 🙏🏻?  
For example:
```
export const hasContractEntered = (contract_info?: TContractInfo) =>
```